File: Sliding Window/BestTimeToBuyAndSellStock.py
def maxProfit(self, prices: List[int]) -> int:
    
    maxProfit = 0
    profit = 0
    
    l = 0
    r = 1
    
    while (r < len(prices)):
        
        if (prices[r] < prices[l]):
            l = r
            
        else:
            profit = prices[r] - prices[l]
            maxProfit = max(profit, maxProfit)
            
        r += 1
        
    return maxProfit

File: 03_Sliding_Window/BestTimeToBuyAndSellStock.java
public class BestTimeToBuyAndSellStock {
    
    public int maxProfit(int[] prices){

        int maxProfit = 0, profit = 0;

        int buy = 0, sell = 1;  // days on which to buy and sell, respectively

        if (prices.length < 2)  
            return 0;

        while (sell < prices.length){

            if (prices[sell] < prices[buy]){

                buy = sell;  

            }

            else{

                profit = prices[sell] - prices[buy];
                maxProfit = Math.max(profit, maxProfit);
                
            }

            sell++;

        }

        return maxProfit;

    }

}
